certbot/certbot/_internal/auth_handler.py

```diff
@@ -284,11 +284,8 @@ def _report_failed_authzrs(self, failed_authzrs):
         for _, achalls in sorted(problems.items(), key=lambda item: item[0]):
             msg.append(_generate_failed_chall_msg(achalls))
 
-        # Show a hint for whatever the first failed challenge was. Every failed challenge will
-        # almost certainly be the same type, except for very exotic --manual setups.
         if failed_achalls:
-            chall_type = failed_achalls[0].typ
-            msg.append('\nHint: {}\n'.format(self.auth.auth_hint(chall_type)))
+            msg.append('\nHint: {}\n'.format(self.auth.auth_hint(failed_achalls)))
```

Comment:
We almost certainly have to check if this is implemented here, since we're not adding this to the interface, just implementing it ourselves.

Reply:
Ah, in which case we should probably document this somewhere, even just in comments on the interface file, because we don't want people implementing it to do something completely else. Or we could just go big and plan to make the change to add it to the interface, and up certbot's major level.